utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -463,7 +472,14 @@ XmlRpcClient::readResponse()
       return false;
     }
     _response += buff;
-
+    // Avoid an overly large response
+    if (_response.length() > size_t(__INT_MAX__)) {
+      XmlRpcUtil::error("XmlRpcClient::readResponse: response length (%u) exceeds the maximum allowed size (%u).",
+                        _response.length(), __INT_MAX__);
+      _response.resize(__INT_MAX__);
```

Comment:
I don't understand this truncation; what will happen to the data in `_response` after this? Will it be garbage-collected? Or passed on to something else for further handling? I suspect discarding all the data would be easier / better than keeping two gigabytes of the data.

Reply:
Changed to clear() in 7d51a7aaea0cf022cc82357e03db6340741364d3